iroh-gossip/src/proto/plumtree.rs

```diff
@@ -0,0 +1,603 @@
+//! Implementation of the Plumtree epidemic broadcast tree protocol
+//!
+//! The implementation is based on [this paper][paper] by Joao Leitao, Jose Pereira, Luıs Rodrigues
+//! and the [example implementation][impl] by Bartosz Sypytkowski
+//!
+//! [paper]: https://asc.di.fct.unl.pt/~jleitao/pdf/srds07-leitao.pdf
+//! [impl]: https://gist.github.com/Horusiath/84fac596101b197da0546d1697580d99
+
+use std::{
+    collections::{HashMap, HashSet, VecDeque},
+    fmt,
+    hash::Hash,
+    time::Duration,
+};
+
+use bytes::Bytes;
+use derive_more::{Add, From, Sub};
+use indexmap::IndexSet;
+use serde::{Deserialize, Serialize};
+
+use super::{PeerAddress, IO};
```

Comment:
Yeah, I though the same, there's this comment already on the `PeerAddress` type:
```
/// TODO: Rename to `PeerId`? It does not necessarily refer to a peer's address, as long as the
/// networking layer can translate the value of its concrete type into an address.
```
I was worried a bit about just renaming to `PeerId` because having the same token for a trait (here) and struct (iroh-net and iroh) might be a bit weird. If you or someone else has suggestions, please share :-)

We could do `PeerIdT`, I've seen it at other places, but also was never too fond of that nomenclature. But it's fine, I guess. Or maybe `PeerIdentity` or something like this.